openapi-cli/src/test/java/org/ballerinalang/openapi/CodeGeneratorTest.java

```diff
@@ -112,13 +138,13 @@ public void generateSkeletonForRequestbody() {
                 expectedServiceContent = (expectedServiceContent.trim()).replaceAll("\\s+", "");
 
                 Assert.assertTrue(generatedService.contains(expectedServiceContent));
-                deleteGeneratedFiles("openapipetstore-service.bal");
             } else {
                 Assert.fail("Service was not generated");
             }
         } catch (IOException | BallerinaOpenApiException e) {
-            deleteGeneratedFiles("openapipetstore-service.bal");
             Assert.fail("Error while generating the service. " + e.getMessage());
+        } finally {
+            deleteGeneratedFiles("openapipetstore-service.bal");
```

Comment:
If the file was not generated, wont this throw an exception according to the `deleteGeneratedFiles` function ?
Use `Files.deleteIfExists()` inside `deleteGeneratedFiles` to avoid this.